src/kernels/kernelProvider.web.ts

```diff
@@ -50,12 +52,12 @@ export class KernelProvider extends BaseKernelProvider {
         const kernel = new Kernel(
             uri,
             resourceUri,
-            options.metadata,
+            KernelConnectionMetadataProxy.wrap(options.metadata),
```

Comment:
Would it make sense to do this inside of the Kernel class?

Reply:
Commented in another place, i think its best left in kernel provider.